homeassistant/components/modbus/validators.py

```diff
@@ -155,12 +155,12 @@ def number_validator(value: Any) -> int | float:
 
     try:
         value = int(value)
-        return value
+        return cast(int, value)
     except (TypeError, ValueError):
         pass
     try:
         value = float(value)
-        return value
+        return cast(float, value)
```

Comment:
No need to use `cast` here.
```suggestion
        return float(value)
```

Reply:
I must have been asleep. Corrected